Support multiple Jenkins instances in config file

radomirbosak opened the pull request:
This commit adds sections to the .jenkins-cli config file. The section
which should be used is specified with the -e or --environment. If no
environment is specified, the "DEFAULT" section is used.

Fixes #36

radomirbosak committed:
Support multiple Jenkins instances in config file

This commit adds sections to the .jenkins-cli config file. The section
which should be used is specified with the -e or --environment. If no
environment is specified, the "DEFAULT" section is used.

Fixes #36

LD250 commented:
Hello @radomirbosak .

I have two suggestions:
1) Make error message more informative. 
For example, running ```jenkins -e LOCAL``` will give me ```No such environment: LOCAL``` error. This looks a bit confusing. I would change it for something like ```LOCAL section in .jencins-cli settings file is not found```

2) I think it is good practice to test cover new functionality. In this case ```TestCliFileUsing``` can be extended with new test case which tests that correct section is used when calling ```JenkinsCli.read_settings_from_file``` with respective environment.

radomirbosak committed:
Make missing config section error message clearer

radomirbosak committed:
Add unit test for reading different config section

radomirbosak committed:
Close the config file properly

radomirbosak commented:
Hey @LD250 , I changed the error message and added a unit test for the new functionality.

The third commit fixes a unclosed-file warning revealed by the unit tests:
```console
$ python3 -m unittest discover tests/
/home/rbosak/repos/jenkins-cli-python/jenkins_cli/cli.py:129: ResourceWarning: unclosed file <_io.TextIOWrapper name='/home/hupu/.jenkins-cli' mode='r' encoding='UTF-8'>
  config.readfp(open(filename, 'r'))
```

please take a look (PTAL)

LD250 reviewed (approved):
Everything is good, thank you!
